gtsam/inference/EliminateableFactorGraph-inst.h

```diff
@@ -36,17 +36,17 @@ namespace gtsam {
       // no Ordering is provided.  When removing optional from VariableIndex, create VariableIndex
       // before creating ordering.
       VariableIndex computedVariableIndex(asDerived());
-      return eliminateSequential(function, computedVariableIndex, orderingType);
+      return eliminateSequential(orderingType, function, computedVariableIndex);
```

Comment:
thanks for the PR, Varun.

Why change the order of variables here? breaks backwards compatibility perhaps?

Reply:
It doesn't because this is a function call within another function aka the public API is still the same.

Fan and I discovered that the original function call was just calling this function, so instead of the indirect call, we save on the call stack by making this call directly.

Reply:
Another way to think of this is that we're just calling another version of an overloaded function, but with the added benefit of reducing the call stack, thus minor gains in efficiency plus easier to read the stack trace during debugging.

Reply:
Cool, makes sense. Could you add a link to the old wrapper function and the new function we're calling directly? Which file are they in?

Reply:
This is the original one which is deprecated (though not properly).
https://github.com/borglab/gtsam/blob/744db328e7ae537e71329e04cc141b3a28b0d6bd/gtsam/inference/EliminateableFactorGraph.h#L301

This is replaced by this https://github.com/borglab/gtsam/blob/744db328e7ae537e71329e04cc141b3a28b0d6bd/gtsam/inference/EliminateableFactorGraph.h#L117

Reply:
I see. So now we are using overloaded versions with boost "none" defaults for missing args. I wonder if we should make a TODO or note to ** fully ** remove the deprecated one as well

Reply:
Yup, that's issue #889

Reply:
We don't do TODOs here, we make issues haha